68 fix footer styling amt

allison-truhlar opened the pull request:
### What changes did you make?
- Changes to the Footer.jsx and Footer.css.

### Why did you make these changes?
- To fix styling at intermediate screen sizes that made the list of designer and developer names display in columns that were difficult to read/didn't look good.
- To make the Footer fixed to the bottom of the viewport

### Screenshots:

<details><summary>Before</summary>

![Screenshot 2023-11-15 at 12 08 52 PM](https://github.com/chingu-voyages/v46-tier2-team-20/assets/114596142/0a104451-9e18-41bf-bda0-ee04cc83320c)

</details> 

<details><summary>After</summary>

![Screenshot 2023-11-15 at 12 09 52 PM](https://github.com/chingu-voyages/v46-tier2-team-20/assets/114596142/7660bca4-9fc2-40bc-be3b-aa5b0ea1b055)

</details>

allison-truhlar committed:
Changes to Footer.jsx and Footer.css

These changes fixed an issue where, at intermediate screen sizes, the developer and designer names went into columns that made it difficult to read.

AOA19 commented on src/components/footer/Footer.jsx:
```diff
@@ -3,46 +3,45 @@ import './Footer.css';
-                <a href="https://github.com/allison-truhlar" target="blank">Allison</a>
-              </span>
+              <a href="https://github.com/allison-truhlar" target="blank">Allison</a>
               {' '}
-              <span>
-                <a href="https://github.com/Emimint" target="blank">Emimint</a>
-              </span>
+              <a href="https://github.com/Emimint" target="blank">Emimint</a>
               {' '}
-              <span>
-                <a href="https://github.com/komalgill0310" target="blank">Komal</a>
-              </span>
+              <a href="https://github.com/komalgill0310" target="blank">Komal</a>
```
@allison-truhlar The value for the `target` attribute should be `_blank`. Right now  when you click on each of our names it doesn't open in a new tab.

AOA19 commented:
@allison-truhlar The footer looks great! I just have one small suggestion, instead of our names being underlined by default maybe they can have an underline effect when hovered over? 

Also I was going to change this later but since you're already working it, do you mind changing my name to match my Github username (AOA19)?

allison-truhlar committed:
Added target="_blank" to all links

allison-truhlar committed:
Changed Alexandra to AOA19

allison-truhlar committed:
Changed the footer from "fixed" to "sticky"

This way, when the page becomes populated with recipes, the footer gets pushed to the very bottom of the recipes, instead of always being at the bottom of the viewport

allison-truhlar committed:
Add underline to links on hover

allison-truhlar commented:
@AOA19 Thanks for these catches/suggestions! 

- I added target="blank" and confirmed all the links open in a new tab. 
- I changed "Alexandra" to "AOA19." 
- I made the links underlined on hover (let me know if you were thinking of something different than what it looks like!) 
- Finally, I noticed that I should have made the footer "sticky," not "fixed," so that it is always at the bottom of the recipe container div, and not overlaid on top of it when it gets filled up with recipe cards.

AOA19 commented:
@allison-truhlar 
- The links look good and the hover effect is exactly what I meant. And thank you for changing my name for me!
- As for the position of the footer, I'm not sure what it looks like on your end, but now that the position changed from `fixed` to `sticky` the footer is now floating when the recipe cards aren't visible. 

- Footer with no search 
<img width="1425" alt="footer-nosearch" src="https://github.com/chingu-voyages/v46-tier2-team-20/assets/66294921/d4fc57e9-a045-497b-ae5b-b28778725de5">

- Footer with search
<img width="1425" alt="footer-search" src="https://github.com/chingu-voyages/v46-tier2-team-20/assets/66294921/362b4242-93b3-4f7b-885d-e5e6cc14c76f">

allison-truhlar commented:
@AOA19 Good catch - I'll work on it!

allison-truhlar committed:
Added styling to App.jsx to make footer stick to bottom

Specifically, added flex, flex-col, and min-h-screen Tailwind classes to the main container div in App.jsx. Then, I added a div around all the content except the Footer. This div I gave the Tailwind class flex-grow, so it will grow to fill, at a minimum, the remaining space required to make the parent div reach the viewport height

allison-truhlar committed:
Removed styling that is no longer required on Footer.jsx

Given the changes made in the last commit to App.jsx, the sticky styling is no longer needed on the footer to make it "stick" to the bottom of the page

allison-truhlar commented:
@AOA19 I think the footer should stick to the bottom of the page correctly now - it looks okay on my end. Could you please double-check? Thanks!

AOA19 reviewed (approved):
@allison-truhlar The footer is sticking to the bottom now. I'll go ahead and merge these changes.

AOA19 committed:
Merge branch 'development' into 68-fix-footer-styling-amt
